test/app-tap/uri.test.lua

```diff
@@ -96,6 +96,76 @@ local function test_parse(test)
     test:is(tostring(error), expected_errmsg, "error message")
 end
 
+local function test_parse_ipv6_format_by_rfc2732(test)
+    -- The literal format of an IPv6 address can contain the following
+    -- characters [0-9;a-f;A-F]. It was discovered in that the
+    -- uri does not support the characters A-F, although it should
+    -- according to the RFC 2732.
+    test:plan(36)
+
+    local u
+
+    u = uri.parse("[FEDC:BA98:7654:3210:FEDC:BA98:7654:3210]")
+    test:is(u.host, 'FEDC:BA98:7654:3210:FEDC:BA98:7654:3210', 'host')
+    test:is(u.ipv6, 'FEDC:BA98:7654:3210:FEDC:BA98:7654:3210', 'ipv6')
+
+    u = uri.parse("[1080:0:0:0:8:800:200C:4171]")
+    test:is(u.host, '1080:0:0:0:8:800:200C:4171', 'host')
+    test:is(u.ipv6, '1080:0:0:0:8:800:200C:4171', 'ipv6')
+
+    u = uri.parse("[3ffe:2a00:100:7031::1]")
+    test:is(u.host, '3ffe:2a00:100:7031::1', 'host')
+    test:is(u.ipv6, '3ffe:2a00:100:7031::1', 'ipv6')
```

Comment:
It is not clear why do you choose these examples for testcases.
The patch fixes support of symbols `[0-9;a-f;A-F]` in `uri.parse`, so I would use one example per supported symbol:

Like this:
- `0`: `uri.parse("[0:0:0:0:0:0:0]")`
- `1`: `uri.parse("[1:1:1:1:1:1:1]")`
- ...
- `f`: `uri.parse("[f:f:f:f:f:f:f]")`
- `F`: `uri.parse("[F:F:F:F:F:F:F]")`

Reply:
These examples are taken from RFC 2732, which is referenced in the issue. Not all IPv6 address variants are supported in tarantool:uri, but we have decided not to expand the supported cases in this patch ([see](https://github.com/tarantool/tarantool/issues/9556#issuecomment-2252481861)).

Reply:
My motivation is the following:
with simple examples, debugging parser would be easier. With failed testcase `uri.parse("[0:0:0:0:0:0:0]")` it is obvious that something is wrong with symbol `0` in a parser and the same for testcases with other supported symbols.

Ok, I'll not insist with this. Feel free to ignore and resolve the comment.